### DISCUSSION TITLE
Pushing masked bitmap onto a sprite

### DISCUSSION THREAD

**Discussion by ExitStatus** [Bodmer/TFT_eSPI#2660] [Q&A - Sprites] [answered]
I have working code taken from the examples that will decode a PNG with transparency directly to the display using the following code

```
    decoder.getLineAsRGB565(pDraw, lineBuffer, PNG_RGB565_BIG_ENDIAN, 0xffffffff);

    if (decoder.getAlphaMask(pDraw, maskBuffer, 255))
        tft.pushMaskedImage(pngDrawX, pngDrawY + pDraw->y, pDraw->iWidth, 1, lineBuffer, maskBuffer);
```

This works fine, but to remove flickering I would like to push this to a sprite which I am using as an offscreen buffer to build part of the display and then push that sprite to the display. Sort of with a "sprite.pushMaskedImage", but this doesn't seem to exist. Is this at all possible?

When I push a masked image to the display it shows up fine, but pushing the masked image to the sprite just nothing is rendered.

**Comment by Bodmer** [comment, marked as answer]
That is not supported by the library, it is on the very long TODO list!

However I have a sketch that does what you want:
[Flash_transparent_PNG_to_sprite.zip](https://github.com/Bodmer/TFT_eSPI/files/11502743/Flash_transparent_PNG_to_sprite.zip)

This contains a copy of the pushMaskedImage library function with minor changes write to a sprite (spr) instead of the TFT.

**Reply by ExitStatus** [reply]
Thanks. That looks great. I will try it out.

**Reply by ExitStatus** [reply]
I dropped the code in, a couple of small tweaks to the code on my side and it worked first time. Perfect.
